### PR TITLE
Fix Crash on Unreadable Gallery Image

### LINKED ISSUE #5909: [ACRA 2.9] BasicImageFieldController onActivityResult Crash
I'll get on this tomorrow. Two issues: 

* Should be able to decode media
* An exception shouldn't crash the viewer.

https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/879cca9f-143b-4db1-ad67-28831d6b223a

```
java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=1, result=-1, data=Intent { dat=content://com.simplemobiletools.gallery.provider/external_files/data/data/com.simplemobiletools.gallery/files/storage/emulated/0/Pictures/Screenshots/[REDACTED] typ=image/png flg=0x1 }} to activity {com.ichi2.anki/com.ichi2.anki.multimediacard.activity.MultimediaEditFieldActivity}: java.lang.IllegalStateException: Couldn't read row 0, col -1 from CursorWindow. Make sure the Cursor is initialized correctly before accessing data from it.
at android.app.ActivityThread.deliverResults(ActivityThread.java:4428)
at android.app.ActivityThread.handleSendResult(ActivityThread.java:4472)
at android.app.ActivityThread.-wrap19(Unknown Source:0)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1686)
at android.os.Handler.dispatchMessage(Handler.java:110)
at android.os.Looper.TinnoLoop(Looper.java:189)
at android.os.Looper.loop(Looper.java:270)
at android.app.ActivityThread.main(ActivityThread.java:6725)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
Caused by: java.lang.IllegalStateException: Couldn't read row 0, col -1 from CursorWindow. Make sure the Cursor is initialized correctly before accessing data from it.
at android.database.CursorWindow.nativeGetString(Native Method)
at android.database.CursorWindow.getString(CursorWindow.java:438)
at android.database.AbstractWindowedCursor.getString(AbstractWindowedCursor.java:51)
at android.database.CursorWrapper.getString(CursorWrapper.java:137)
at com.ichi2.anki.multimediacard.fields.BasicImageFieldController.onActivityResult(BasicImageFieldController.java:6)
at com.ichi2.anki.multimediacard.activity.MultimediaEditFieldActivity.onActivityResult(MultimediaEditFieldActivity.java:3)
at android.app.Activity.dispatchActivityResult(Activity.java:7403)
at android.app.ActivityThread.deliverResults(ActivityThread.java:4424)
... 10 more
```

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description

Crash caused due to being unable to decode the following intent.

```
Failure delivering result ResultInfo{who=null, request=1, result=-1, 
data=Intent { dat=content://com.simplemobiletools.gallery.provider/external_files/data/data/com.simplemobiletools.gallery/files/storage/emulated/0/Pictures/Screenshots/[REDACTED] typ=image/png flg=0x1 }}
 to activity {com.ichi2.anki/com.ichi2.anki.multimediacard.activity.MultimediaEditFieldActivity}: java.lang.IllegalStateException: Couldn't read row 0, col -1 from CursorWindow. Make sure the Cursor is initialized correctly before accessing data from it.
```

## Fixes
Fixes #5909 

## Approach
We catch any exceptions, or show a toast on known failure

## How Has This Been Tested?

Unit Test. Couldn't repro on my device. Image insert still works correctly

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- AnkiDroid/src/test/java/com/ichi2/testutils/MockContentResolver.java  1
- AnkiDroid/src/test/java/com/ichi2/testutils/MockCursor.java  1
- AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicImageFieldController.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5301014&bid=17351138)

File: AnkiDroid/src/test/java/com/ichi2/anki/multimediacard/activity/MultimediaEditFieldActivityTestBase.java [hunk 1]
```diff
@@ -10,13 +10,17 @@
...
 
+import org.mockito.Mockito;
 import org.robolectric.Robolectric;
 import org.robolectric.Shadows;
 import org.robolectric.android.controller.ActivityController;
 import org.robolectric.shadows.ShadowApplication;
 
 import androidx.test.core.app.ApplicationProvider;
 
+import static org.mockito.ArgumentMatchers.matches;
```
Comment (timrae) [inline AnkiDroid/src/test/java/com/ichi2/anki/multimediacard/activity/MultimediaEditFieldActivityTestBase.java:21]: Codacy Issue found: [Avoid unused imports such as 'org.mockito.ArgumentMatchers.matches'](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5301014)

File: AnkiDroid/src/test/java/com/ichi2/anki/multimediacard/fields/BasicImageFieldControllerTest.java [hunk 2]
```diff
@@ -65,6 +72,24 @@ public void erroringFileDoesNotDisplayPreview() {
                 is(false));
     }
 
+
+    @Test
+    public void invalidImageResultDoesNotCrashController() {
```
Comment (timrae) [inline AnkiDroid/src/test/java/com/ichi2/anki/multimediacard/fields/BasicImageFieldControllerTest.java:77]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5301014)

File: AnkiDroid/src/test/java/com/ichi2/testutils/MockCursor.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+import android.database.Cursor;
+import android.database.CursorWrapper;
+
+import org.mockito.Mockito;
+
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.when;
+
+public class MockCursor {
```
Comment (timrae) [inline AnkiDroid/src/test/java/com/ichi2/testutils/MockCursor.java:11]: Codacy Issue found: [The utility class name 'MockCursor' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5301014)

File: AnkiDroid/src/test/java/com/ichi2/testutils/MockContentResolver.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+import android.content.ContentResolver;
+import android.database.Cursor;
+
+import org.mockito.Mockito;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+public class MockContentResolver {
```
Comment (timrae) [inline AnkiDroid/src/test/java/com/ichi2/testutils/MockContentResolver.java:11]: Codacy Issue found: [The utility class name 'MockContentResolver' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5301014)

File: AnkiDroid/src/test/java/com/ichi2/anki/multimediacard/activity/MultimediaEditFieldActivityTestBase.java (hunk 1 above)
Comment (mikehardy) [reply AnkiDroid/src/test/java/com/ichi2/anki/multimediacard/activity/MultimediaEditFieldActivityTestBase.java]: I made this so I could check things locally - I diff the sorted contents of a recent master run vs branch run and my issues stand out pretty well
https://github.com/ankidroid/Anki-Android/blob/master/tools/quality-check/codacy-ankidroid.sh
If you're interested let me know and I'll get you a codacy API key

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- AnkiDroid/src/test/java/com/ichi2/testutils/MockCursor.java  1
- AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicImageFieldController.java  2
- AnkiDroid/src/test/java/com/ichi2/testutils/MockContentResolver.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5301014&bid=17351138)

File: AnkiDroid/src/test/java/com/ichi2/anki/multimediacard/fields/BasicImageFieldControllerTest.java (hunk 2 above)
Comment (timrae) [inline AnkiDroid/src/test/java/com/ichi2/anki/multimediacard/fields/BasicImageFieldControllerTest.java:77]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5301014)

File: AnkiDroid/src/test/java/com/ichi2/testutils/MockCursor.java (hunk 3 above)
Comment (timrae) [inline AnkiDroid/src/test/java/com/ichi2/testutils/MockCursor.java:11]: Codacy Issue found: [The utility class name 'MockCursor' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5301014)

File: AnkiDroid/src/test/java/com/ichi2/testutils/MockContentResolver.java (hunk 4 above)
Comment (timrae) [inline AnkiDroid/src/test/java/com/ichi2/testutils/MockContentResolver.java:11]: Codacy Issue found: [The utility class name 'MockContentResolver' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5301014)

File: AnkiDroid/src/test/java/com/ichi2/anki/multimediacard/fields/BasicImageFieldControllerTest.java (hunk 2 above)
Comment (david-allison-1) [reply AnkiDroid/src/test/java/com/ichi2/anki/multimediacard/fields/BasicImageFieldControllerTest.java:77]: It does

**Comment by david-allison-1** [comment]
I'll leave the Codacy reviews to a human, they look OK to me, but I'm not opposed to changes.

**Comment by david-allison-1** [comment]
LGTM

<p align="center">
  
</p>

**Pull request merged**